### DISCUSSION TITLE
Worker Failing to Pull Certs from Full Node

### DISCUSSION THREAD

**Discussion by nonlinearsugar** [guydavis/machinaris#280] [Q&A] [answered]
I've used the new worker wizard on my full Machinaris node server to generate a config for a worker on a Windows 10 machine running Docker CE on the same subnet as the full node, and am getting this error each time I load the container:

`Certificates response of 000 from http://shale:8927/certificates/?type=chia. Try clicking 'New Worker' button on 'Workers' page first.`

The worker can resolve the full node by hostname, and the full node can resolve the worker by hostname.

The worker can access ports 8926 and 8927 on full node, and the full node can access ports 8926 and 8927 on the worker node.

Using a web browser on the worker, I'm able to access the URL I posted above, and I get a certs.zip package.

The worker is not showing up in the Workers list in the main Machinaris console.

Both nodes are running 0.5.4.

Please assist - how do I get the worker up and running?

**Comment by guydavis** [comment, marked as answer]
Sorry to hear that.  Please paste your full Docker run/compose here for review.  My recommendation will be:

1. Use an IP address for `controller_host`, not a hostname.
2. OR pass your DNS server's IP address.

Both are fields in the New Worker wizard.  Vast majority of Machinaris users chose #1, to specify IP addresses to get around local DNS issues (particularly the interaction with Docker networking).  

My guess is that in-container is responding differently than your host OS.  To check, click the New Worker button on Workers page, then start the Machinaris container, then:
```
docker exec -it machinaris bash
ping shale
curl -v http://shale:8927/certificates/?type=chia
```
Hope this helps!

**Comment by nonlinearsugar** [comment]
Switching controller_host to IP address instead of hostname worked.  For some reason, shale was resolving to 127.0.1.1.  I don't have nslookup in the container and setting it to IP isn't necessarily a bad thing so I'll stick with that solution.

Thank you so much for the prompt response!  I had been banging my head against this for four hours.

May I suggest you indicate the vast majority of users specify by IP address instead of hostname due to negative interactions with docker networking in the worker setup docs?  Just to hopefully save others the same time.

Thanks for your work.

**Reply by guydavis** [reply]
Glad to hear it helped.  I enjoy working on free software projects.  No donations needed.
